CFGInventoryFormatter_pass.py::test_add_ips_to_groups FAILED

Running `python setup.py test` fails on test `CFGInventoryFormatter_pass.py::test_add_ips_to_groups FAILED` with:

```
___________________________________________________________________________________ test_add_ips_to_groups ____________________________________________________________________________________

    @with_setup(setup_json_inventory_formatter)
    def test_add_ips_to_groups():
        """
        """
        inven_hosts = inv['hosts']
        formatter.add_sections(inv['host_groups'].keys())
>       formatter.add_ips_to_groups(inven_hosts, inv)

linchpin/tests/InventoryFilters/test_CFGInventoryFormatter_pass.py:102:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
linchpin/InventoryFilters/CFGInventoryFormatter.py:55: in add_ips_to_groups
    self.add_ips_to_host_group("all", inven_hosts)
linchpin/InventoryFilters/CFGInventoryFormatter.py:74: in add_ips_to_host_group
    self.config.set(host_group, ip)
../../../.pyenv/versions/linchpin/lib/python2.7/site-packages/backports/configparser/__init__.py:1238: in set
    _, option, value = self._validate_value_types(option=option, value=value)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <backports.configparser.ConfigParser object at 0x7fef763f9d50>, kwargs = {'option': {'count': 2, 'host_groups': ['nodes', 'glusterfs'], 'name': '-node'}, 'value': None}, section = ''
option = {'count': 2, 'host_groups': ['nodes', 'glusterfs'], 'name': '-node'}, value = None

    def _validate_value_types(self, **kwargs):
        """Raises a TypeError for non-string values.

        The only legal non-string value if we allow valueless
        options is None, so we need to check if the value is a
        string if:
        - we do not allow valueless options, or
        - we allow valueless options but the value is not None

        For compatibility reasons this method is not used in classic set()
        for RawConfigParsers. It is invoked in every case for mapping protocol
        access and in ConfigParser.set().
        """
        # keyword-only arguments
        section = kwargs.get('section', "")
        option = kwargs.get('option', "")
        value = kwargs.get('value', "")

        if PY2 and bytes in (type(section), type(option), type(value)):
            # we allow for a little unholy magic for Python 2 so that
            # people not using unicode_literals can still use the library
            # conveniently
            warnings.warn(
                "You passed a bytestring. Implicitly decoding as UTF-8 string."
                " This will not work on Python 3. Please switch to using"
                " Unicode strings across the board.",
                DeprecationWarning,
                stacklevel=2,
            )
            if isinstance(section, bytes):
                section = section.decode('utf8')
            if isinstance(option, bytes):
                option = option.decode('utf8')
            if isinstance(value, bytes):
                value = value.decode('utf8')

        if not isinstance(section, str):
            raise TypeError("section names must be strings")
        if not isinstance(option, str):
>           raise TypeError("option keys must be strings")
E           TypeError: option keys must be strings

../../../.pyenv/versions/linchpin/lib/python2.7/site-packages/backports/configparser/__init__.py:1218: TypeError
```

--- a/linchpin/tests/InventoryFilters/test_CFGInventoryFormatter_pass.py
+++ b/linchpin/tests/InventoryFilters/test_CFGInventoryFormatter_pass.py
@@ -32,7 +32,9 @@ def setup_json_inventory_formatter():
     template = 'parsed-layout.json'
     template_file = open(mock_path+'/'+template)
     inv = json.load(template_file)
-
+    # this converts everything from unicode to str.  Can be deleted
+    # once python2 support is dropped
+    # inv = json.dumps(json.loads(inv))
 
 def setup_cfg_config():
     global config
@@ -98,8 +100,10 @@ def test_set_vars():
 def test_add_ips_to_groups():
     """
     """
-    inven_hosts = inv['hosts']
-    formatter.add_sections(list(inv['host_groups'].keys()))
+    inven_hosts = []
+    for host in inv['hosts']:
+        inven_hosts.append(host['name'])
+    formatter.add_sections(inv['host_groups']['OSEv3']['children'])
     formatter.add_ips_to_groups(inven_hosts, inv)
  
 

--- a/linchpin/tests/InventoryFilters/test_InventoryFilter_pass.py
+++ b/linchpin/tests/InventoryFilters/test_InventoryFilter_pass.py
@@ -27,7 +27,25 @@ def setup_inventory_filter():
 
     template = 'parsed-layout.json'
     template_file = open(mock_path+'/'+template)
-    inv = json.load(template_file)
+    # object hook can be removed once python 2 is no longer supported
+    inv = json.load(template_file, object_hook=_byteify)
+
+def _byteify(data, ignore_dicts = False):
+    # if this is a unicode string, return its string representation
+    if isinstance(data, unicode):
+        return data.encode('utf-8')
+    # if this is a list of values, return list of byteified values
+    if isinstance(data, list):
+        return [ _byteify(item, ignore_dicts=True) for item in data ]
+    # if this is a dictionary, return dictionary of byteified keys and values
+    # but only if we haven't already byteified it
+    if isinstance(data, dict) and not ignore_dicts:
+        return {
+            _byteify(key, ignore_dicts=True): _byteify(value, ignore_dicts=True)
+            for key, value in data.iteritems()
+        }
+    # if it's anything else, return it in its original form
+    return data
 
 @with_setup(setup_inventory_filter)
 def test_get_layout_hosts():
@@ -94,7 +112,10 @@ def test_add_ips_to_groups():
     """
     I don't know what this is
     """
-    inven_hosts = inv['hosts']
+    inven_hosts = []
+    for host in inv["hosts"]:
+        inven_hosts.append(host['name'])
+    filter.add_sections(inv['host_groups']['OSEv3']['children'])
     filter.add_ips_to_groups(inven_hosts, inv)
 
 @with_setup(setup_inventory_filter)

--- a/linchpin/tests/InventoryFilters/test_JSONInventoryFormatter_pass.py
+++ b/linchpin/tests/InventoryFilters/test_JSONInventoryFormatter_pass.py
@@ -75,7 +75,7 @@ def test_add_ips_to_groups():
     """
     """
     inven_hosts = inv['hosts']
-    formatter.add_sections(list(inv['host_groups'].keys()))
+    formatter.add_sections(list(inv['host_groups']['OSEv3']['children']))
     formatter.add_ips_to_groups(inven_hosts, inv)
  
 
@@ -87,7 +87,7 @@ def test_add_common_vars():
     formatter.add_sections(host_groups)
     formatter.set_vars(inv)
     formatter.add_ips_to_groups(host_groups, inv)
-    formatter.add_common_vars(host_groups, inv)
+    formatter.add_common_vars(host_groups, inv, {})
 
 
 @with_setup(setup_json_inventory_formatter)

--- a/linchpin/tests/mockdata/complex-inventory/correct-inventory
+++ b/linchpin/tests/mockdata/complex-inventory/correct-inventory
@@ -26,10 +26,10 @@ test-79b201-0 hostname=test-79b201-0
 test-79b201-1 hostname=test-79b201-1
 
 [all]
-test-79b201-2 hostname=test-79b201-2
-test-79b201-0 hostname=test-79b201-0
-test-79b201-1 hostname=test-79b201-1
-10.138.166.226 hostname=10.138.166.226
-intel-sugarbay-dh-02.ml3.eng.bos.redhat.com hostname=intel-sugarbay-dh-02.ml3.eng.bos.redhat.com
 nec-em9.rhts.eng.bos.redhat.com hostname=nec-em9.rhts.eng.bos.redhat.com
+intel-sugarbay-dh-02.ml3.eng.bos.redhat.com hostname=intel-sugarbay-dh-02.ml3.eng.bos.redhat.com
+10.138.166.226 hostname=10.138.166.226
+test-79b201-1 hostname=test-79b201-1
+test-79b201-0 hostname=test-79b201-0
+test-79b201-2 hostname=test-79b201-2
 

